src/types.ts

```diff
@@ -12,12 +12,19 @@ export interface ProviderContext {
 
 export type FontStyles = 'normal' | 'italic' | 'oblique'
 
+// Only registered axis tags are listed.
+type VariableAxis = 'opsz' | 'slnt' | 'wdth' | string
+
 export interface ResolveFontOptions {
   weights: string[]
   styles: FontStyles[]
   // TODO: improve support and support unicode range
   subsets: string[]
   fallbacks?: string[]
+  // Variable axis
+  experimental?: {
+    variableAxis?: { [key: VariableAxis]: string[] }
+  }
```

Comment:
as these currently only work for `google` what about keeping them (for now) in the google provider options? This does mean it's not possible to set them on a per-font basis but - what do you think? 🤷  (Or maybe we can enable them that way, and then support a more generic way of enabling/disabling them in `ResolveFontOptions`.)

Reply:
I think we can construct `ProviderOption` for google in this way so that we can set them on a per-font basis:

```typescript
interface ProviderOption {
  experimental: {
    variableAxis?: {
      // font-name: { axis: ['400', '600', '800..1000'] }
      [key: string]: { [key: VariableAxis]: string[] }
    }
  }
}
```

Reply:
But still we'll need to carry out a unified way to configure the variable axes once the built-in subsetter is out. Google's format seems viable and clean. So maybe we can still keep it in the `ResolveFontFaceOptions` but add warnings if there is a `variableAxis` configured for a non-google font?

Reply:
I'm mostly trying to keep possible breaking change surface as small as possible, until we have at least one second variable provider

(and then, if suitable, I'm very open to your suggestion for `ResolveFontFaceOptions`)

so let's use your format for `ProviderOption` but don't support passing them in resolve options, so there's only one option to configure 🙏

Reply:
@danielroe 

Sure! I've now moved the option to google's provider options.